File: serial/python/serial_r.py
# pip install pyserial
import serial

dev_name = "/dev/tty.usbmodem14201"
speed = 9600
port = serial.Serial(dev_name, speed)

def readline():
    line = ''
    while True:
        ch = port.read().decode()
        line +=ch
        if ch =='\n' or ch == '\r' or ch == '':
            if len(line) > 0 :
                return line
    
try:
    if port.isOpen() :
        print('Serial is ready.')
        while True:
            print("recv:" + port.readline().decode()) 
finally:
    port.close()

File: serial/nodeJs/serial_r.js
// npm install serialport
const SerialPort = require("serialport");
const readline = require('readline');

var dev_name = "/dev/tty.usbmodem14201"
var speed = 9600;

var port = new SerialPort(dev_name, {baudRate:speed}, (err)=>{
    if(err){
        console.log("fail to open", err);
    }else{
        console.log("serial open successfully.");
        var line = readline.createInterface(port);
        line.on('line', (data)=>{
            console.log(data);
        });
    }
});

